mgmtfn/dockplugin/dockplugin.go

```diff
@@ -74,21 +78,28 @@ func InitDockPlugin() error {
 		s.HandleFunc(dispatchPath, logHandler(dispatchPath, dispatchFunc))
 	}
 
-	driverPath := path.Join(pluginPath, driverName) + ".sock"
-	os.Remove(driverPath)
-	os.MkdirAll(pluginPath, 0700)
+	if useUnixSocket {
+		driverPath := path.Join(pluginPath, driverName) + ".sock"
+		os.Remove(driverPath)
+		os.MkdirAll(pluginPath, 0700)
 
-	go func() {
-		l, err := net.ListenUnix("unix", &net.UnixAddr{Name: driverPath, Net: "unix"})
-		if err != nil {
-			panic(err)
-		}
+		go func() {
+			l, err := net.ListenUnix("unix", &net.UnixAddr{Name: driverPath, Net: "unix"})
+			if err != nil {
+				panic(err)
+			}
+
+			log.Infof("docker plugin listening on %s", driverPath)
+			http.Serve(l, router)
+			l.Close()
```

Comment:
I think the `l.Close()` should be in a `defer` on line 91 instead of here. I know this wasn't a part of this patch, but it seems like the right time to fix it.